ui/src/utils/index.ts

```diff
@@ -107,6 +107,11 @@ export const preprocessInput = (data: string, config: Partial<ILunaConfig>) => {
       data = String.fromCharCode(AsciiBackspace);
     }
   }
+  if (config.ctrlCAsCtrlZ === '1') {
+    if (data.charCodeAt(0) === AsciiCtrlC) {
+      data = String.fromCharCode(AsciiCtrlZ);
+    }
+  }
   if (data.includes('\u001b[200~') || data.includes('\u001b[201~')) {
     return data.replace(/\u001b\[200~|\u001b\[201~/g, '');
   }
```

Comment:
There appear to be no significant differences between the two code snippets you provided. Both are correct examples of TypeScript code with a few minor adjustments that seem necessary for completeness or consistency:

- In line 10, `const AsciiBackspace` should likely refer specifically to `\u001b[A`, not using `AsciiBackspace`.
- Line 24 is redundant since it's already checked inside the conditional block.

However, given their current context and without additional information about these specific lines of code used within your project, I cannot make further optimizations recommendations. If there are more specific issues or needs, please provide detailed descriptions.